api/tests/src/db/models/factories.py

```diff
@@ -1821,3 +1821,64 @@ class Meta:
     last_notified_at = factory.Faker("date_time_between", start_date="-5y", end_date="-3y")
 
     searched_opportunity_ids = factory.LazyAttribute(lambda _: random.sample(range(1, 1000), 5))
+
+
+class OpportunityCompetitionFactory(BaseFactory):
+    class Meta:
+        model = opportunity_models.OpportunityCompetition
+
+    opportunity = factory.SubFactory(OpportunityFactory)
+    opportunity_id = factory.LazyAttribute(lambda o: o.opportunity.opportunity_id)
+
+    competition_id = Generators.UuidObj
+    legacy_competition_id = sometimes_none(factory.Faker("random_int", min=1, max=15))
+    public_competition_id = sometimes_none(factory.Faker("random_int", min=1, max=15))
+    legacy_package_id = sometimes_none(factory.Faker("random_int", min=1, max=15))
+
+    competition_title = sometimes_none(factory.Faker("sentence"))
+
+    opening_date = factory.Faker("date_between", start_date="-3w", end_date="-1d")
+    closing_date = factory.LazyAttribute(
+        lambda o: fake.date_time_between(start_date=o.opening_date)
+    )
+
+    grace_period = factory.Faker("random_int", min=1, max=10)
+    contact_info = sometimes_none(factory.Faker("sentence"))
+
+    created_at = factory.Faker("date_time_between", start_date="-5y", end_date="-3y")
+    updated_at = factory.LazyAttribute(
+        lambda o: fake.date_time_between(start_date=o.created_at, end_date="-1y")
+    )
+
+
+class OpportunityCompetitionInstructionFactory(BaseFactory):
+    class Meta:
+        model = opportunity_models.OpportunityCompetitionInstruction
+
+    opportunity_competition = factory.SubFactory(OpportunityCompetitionFactory)
+    competition_id = factory.LazyAttribute(lambda o: o.opportunity_competition.competition_id)
+    competition_instruction_id = Generators.UuidObj
+    file_location = "file_location"
+
+
+class OpportunityApplicationForm(BaseFactory):
+    class Meta:
+        model = opportunity_models.OpportunityApplicationForm
+
+    form_id = Generators.UuidObj
+    form_name = factory.Faker("sentence")
+    form_version = factory.Faker("sentence")
+    is_active = False
+    description = factory.Faker("sentence")
+    agency_code_id = factory.Faker("sentence")
```

Comment:
Some suggestions:
* `form_name` -> `bs` might be a good one: https://faker.readthedocs.io/en/master/providers/faker.providers.company.html#faker.providers.company.Provider.bs
* `form_version` -> I'd probably say we should make this formatted like `1.<random int>` - could randomize both, don't think it matters much
* `is_active` -> default to true, if it's inactive, we'll probably disallow the form for use in certain ways
* `agency_code_id` -> Just reuse the `agency_code` faker provider we made
*